docs/gvisor.md

```diff
@@ -0,0 +1,69 @@
+## gVisor Addon
+[gVisor](https://github.com/google/gvisor/blob/master/README.md), a sandboxed container runtime, allows users to securely run pods with untrusted workloads within Minikube.
+
+### Starting Minikube
+gVisor depends on the containerd runtime to run in Minikube.
+When starting minikube, specify the following flags, along with any additional desired flags:
+
+```shell
+$ minikube start --container-runtime=containerd  \
+    --docker-opt containerd=/var/run/containerd/containerd.sock \
+    --network-plugin=cni
+```
+
+### Enabling gVisor
+To enable this addon, simply run:
+
+```
+$ minikube addons enable gvisor
```

Comment:
What happens if you try to enable the addon without setting the container-runtime to containerd? Does it fail sanely?

Reply:
I added a validation check, so now if a user tries to run `minikube addons enable gvisor` without containerd it will print out an error message.